entry_types/scrolled/app/models/pageflow_scrolled/chapter.rb

```diff
@@ -3,6 +3,30 @@ class Chapter < Pageflow::ApplicationRecord
     include Pageflow::SerializedConfiguration
     include Pageflow::AutoGeneratedPermaId
 
-    has_many :sections, -> { order('position ASC') }, dependent: :destroy, inverse_of: :chapter
+    belongs_to :storyline, touch: true
+    has_many :sections,
+             -> { order('pageflow_scrolled_sections.position ASC') },
+             dependent: :destroy,
+             inverse_of: :chapter
+    has_many :content_elements, through: :sections
+
+    delegate :entry, to: :storyline
+
+    attr_accessor :revision # used on :create to lazily create storyline
+    before_validation :ensure_storyline, on: :create
+    before_create :set_initial_position
+
+    private
+
+    def ensure_storyline
+      unless Storyline.all_for_revision(revision).exists?
+        Storyline.create!(revision: revision, configuration: {main: true})
+      end
+      self.storyline = Storyline.all_for_revision(revision).first
+    end
+
+    def set_initial_position
+      self.position = storyline.chapters.count
+    end
```

Comment:
Instead of setting the position here, I'd suggest following the same pattern as in the paged REST controllers and add `position` to the list of permitted params. That way the client side code can [determine the initial position](https://github.com/codevise/pageflow/blob/d5fd99fa09503bc3aa3d9326b242e7723109cfa1/packages/pageflow/src/editor/models/Storyline.js#L92). In the current editor, the sitemap uses this [to insert chapters](https://github.com/codevise/pageflow-sitemap/blob/a5907142794564b21d3149c907d8b1a27cf847e7/app/assets/javascripts/pageflow/sitemap/editor/controllers/editor_mode_controller.js#L252) and pages at arbitrary positions.